exp/api/v1beta1/machinepool_types.go

```diff
@@ -27,6 +27,10 @@ import (
 const (
 	// MachinePoolFinalizer is used to ensure deletion of dependencies (nodes, infra).
 	MachinePoolFinalizer = "machinepool.cluster.x-k8s.io"
+
+	// MachinePoolNameLabel is the label indicating the name of the MachinePool a Machine is controleld by.
+	// Note: The value of this label may be a hash if the MachinePool name is longer than 63 characters.
```

Comment:
would it make sense for it to always be a hash so it's consistent?

Reply:
I copied that to be consistent with the existing pattern for labels like these. Also it would help that it's not a hash so it's user readable.

```
// MachineSetNameLabel is the label set on machines if they're controlled by MachineSet.
// Note: The value of this label may be a hash if the MachineSet name is longer than 63 characters.
MachineSetNameLabel = "cluster.x-k8s.io/set-name"

// MachineDeploymentNameLabel is the label set on machines if they're controlled by MachineDeployment.
MachineDeploymentNameLabel = "cluster.x-k8s.io/deployment-name"

// MachineControlPlaneNameLabel is the label set on machines if they're controlled by a ControlPlane.
// Note: The value of this label may be a hash if the control plane name is longer than 63 characters.
MachineControlPlaneNameLabel = "cluster.x-k8s.io/control-plane-name"
```